src/server/rdb_load.cc

```diff
@@ -1770,6 +1772,16 @@ error_code RdbLoader::Load(io::Source* src) {
       continue;
     }
 
+    // TODO: Shouldn't this all be a switch really?
+    if (type == RDB_OPCODE_JOURNAL_BLOB) {
+      // We should flush all changes on the current db before applying incremental changes.
```

Comment:
why do we need flushing? trying to understand the logic.

Reply:
Because before the flush we have data accumulated inside those per-shard buffers.

It might happen so that some journal command will work with a key that is inside the buffers, but it must be executed strictly after the rdb buffer is loaded